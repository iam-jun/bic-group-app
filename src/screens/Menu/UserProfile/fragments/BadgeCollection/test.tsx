import React from 'react';
import {
  View, Dimensions, FlatList, StyleSheet,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import Avatar from '~/baseComponents/Avatar';
import Icon from '~/baseComponents/Icon';
import spacing, { borderRadius } from '~/theme/spacing';

const Grid = ({ data }: {data: any[]}) => {
  const deviceWidth = Dimensions.get('window').width;

  const itemWidth = 48; // Constant item width
  const numColumns = Math.floor((deviceWidth - 10 * 2) / (itemWidth + 6 * 2));
  const rowWidth = numColumns * (itemWidth + 6 * 2);
  const paddingLeft = (deviceWidth - (10 * 2) - rowWidth) / 2;

  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const styles = themeStyles(theme);

  const isSelected = false;

  const renderGridItem = ({ item, index }: any) =>
    // Render each grid item here
    (
      <View
        key={index}
        style={{
          margin: 6,
          borderWidth: isSelected ? 2 : 1,
          borderColor: isSelected ? colors.purple50 : colors.neutral5,
          borderRadius: borderRadius.pill,
        }}
      >
        <Avatar.Medium
          isRounded
          source={{ uri: item?.iconUrl }}
        />
        {isSelected
          ? (
            <View style={styles.iconChose}>
              <Icon size={10} icon="Check" tintColor={colors.white} />
            </View>
          )
          : null}
      </View>
    );

  return (
    <View style={{
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      paddingHorizontal: 10,
      paddingBottom: 10,
    }}
    >

      <View style={{
        // paddingLeft,
      }}
      >
        <FlatList
          data={data}
          renderItem={renderGridItem}
          keyExtractor={(item, index) => index.toString()}
          numColumns={numColumns}
        />
      </View>
    </View>
  );
};

const themeStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      flex: 1,
      paddingTop: spacing.padding.large,
    },
    iconChose: {
      backgroundColor: colors.purple50,
      borderRadius: borderRadius.pill,
      height: 20,
      width: 20,
      alignItems: 'center',
      justifyContent: 'center',
      position: 'absolute',
      zIndex: 1,
      right: 0,
    },
  });
};

export default Grid;
