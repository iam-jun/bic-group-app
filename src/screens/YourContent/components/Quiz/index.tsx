import React, { useEffect } from 'react';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import {
  StyleSheet, RefreshControl, View, ActivityIndicator,
} from 'react-native';
import Animated from 'react-native-reanimated';
import Image from '~/components/Image';
import images from '~/resources/images';
import Text from '~/baseComponents/Text';
import { homeHeaderTabHeight } from '~/theme/dimension';
import useDraftQuizStore from './store';
import { spacing } from '~/theme';
import ViewSpacing from '~/beinComponents/ViewSpacing';
import QuizPostView from './components/QuizPostView';

interface QuizProps {
    onScroll: (e: any) => void;
}

const Quiz: React.FC<QuizProps> = ({ onScroll }) => {
  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const { draftQuiz, actions } = useDraftQuizStore();
  const {
    data,
    loading,
    refreshing,
    hasNextPage,
  } = draftQuiz || {};

  useEffect(() => {
    getData(true);
  }, []);

  const getData = (isRefresh?: boolean) => {
    actions.getDraftQuiz(isRefresh);
  };

  const onRefresh = () => {
    getData(true);
  };

  const onLoadMore = () => {
    if (hasNextPage) {
      getData(false);
    }
  };

  const renderItem = ({ item }) => (<QuizPostView data={item} />);

  const renderHeader = () => (
    <View style={styles.header}>
      <ViewSpacing height={spacing.margin.large} />
    </View>
  );

  const renderFooter = () => {
    if (!loading) return <ViewSpacing height={spacing.padding.large} />;

    return (
      <View style={styles.boxFooter}>
        <ActivityIndicator />
      </View>
    );
  };

  const renderSeparatorComponent = () => <ViewSpacing height={spacing.margin.large} />;

  const renderEmpty = () => {
    if (hasNextPage) return null;

    return (
      <View style={styles.boxEmpty} testID="draft_quiz.empty_view">
        <Image
          resizeMode="contain"
          source={images.img_empty_box}
          style={styles.imgEmpty}
        />
        <Text.BodyS color={colors.neutral40} useI18n>
          quiz:text_no_quiz_content
        </Text.BodyS>
      </View>
    );
  };

  const keyExtractor = (item) => `draft-quiz-${item?.id}`;

  return (
    <Animated.FlatList
      testID="quiz.list"
      data={data}
      renderItem={renderItem}
      keyExtractor={keyExtractor}
      ListHeaderComponent={renderHeader}
      ListFooterComponent={renderFooter}
      ItemSeparatorComponent={renderSeparatorComponent}
      ListEmptyComponent={renderEmpty}
      refreshControl={(
        <RefreshControl
          refreshing={refreshing}
          onRefresh={onRefresh}
          tintColor={colors.gray40}
          progressViewOffset={homeHeaderTabHeight}
        />
            )}
      onEndReached={onLoadMore}
      onEndReachedThreshold={0.2}
      onScroll={onScroll}
    />
  );
};

const styles = StyleSheet.create({
  boxEmpty: {
    alignItems: 'center',
    paddingTop: 32,
    paddingBottom: 48,
  },
  imgEmpty: {
    width: 100,
    aspectRatio: 1,
    marginBottom: spacing.margin.base,
  },
  header: {
    paddingTop: homeHeaderTabHeight,
  },
  boxFooter: {
    height: 100,
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default Quiz;
