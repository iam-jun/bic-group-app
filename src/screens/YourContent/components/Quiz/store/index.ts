import { createStore, resetStore } from '~/store/utils';
import IBaseState, { InitStateType } from '~/store/interfaces/IBaseState';
import { IPost } from '~/interfaces/IPost';
import getDraftQuiz from './actions/getDraftQuiz';

export interface IDraftQuizState extends IBaseState {
    draftQuiz: {
        data: IPost[];
        loading: boolean;
        refreshing: boolean,
        hasNextPage: boolean,
        endCursor: string,
    },

    actions: {
        getDraftQuiz: (isRefresh?: boolean) => void;
    }
}

const initState: InitStateType<IDraftQuizState> = {
    draftQuiz: {
        data: [],
        loading: false,
        refreshing: false,
        hasNextPage: true,
        endCursor: '',
    },
};

const draftQuizStore = (set, get) => ({
    ...initState,

    actions: {
        getDraftQuiz: getDraftQuiz(set, get),
    },

    reset: () => resetStore(initState, set),
});

const useDraftQuizStore = createStore<IDraftQuizState>(draftQuizStore);
export default useDraftQuizStore;
