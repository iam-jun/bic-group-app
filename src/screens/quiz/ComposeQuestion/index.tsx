import React, { FC, useRef } from 'react';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import { StyleSheet, View } from 'react-native';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import { FormProvider, useFieldArray } from 'react-hook-form';
import { isNumber } from 'lodash';
import useQuizzesStore from '~/store/entities/quizzes';
import Header from '~/beinComponents/Header';
import { useBaseHook } from '~/hooks';
import useComposeQuestion, {
  ValidateComposeQuestionError,
} from './hooks/useComposeQuestion';
import { spacing } from '~/theme';
import QuestionField from './components/QuestionField';
import AnswerField from './components/AnswerField';
import { useBackPressListener } from '~/hooks/navigation';
import { MAX_ANSWERS } from '../helper';
import { Button } from '~/baseComponents';
import Icon from '~/baseComponents/Icon';
import Text from '~/baseComponents/Text';

type ComposeQuestionProps = {
  route?: {
    params?: {
      questionIndex?: number;
      quizId: string;
    };
  };
};

const ComposeQuestion: FC<ComposeQuestionProps> = (props) => {
  const { route } = props;
  const { params } = route || {};
  const { questionIndex, quizId } = params || {};
  const isCreatingQuestion = !isNumber(questionIndex);

  const scrollViewRef = useRef<any>();

  const loading = useQuizzesStore((state) => state.loading);
  const quiz = useQuizzesStore((state) => state.data[quizId]);
  const { questions = [] } = quiz || {};

  const {
    methods,
    enabledBtnSave,
    needToChooseCorrectAnswer,
    onSave,
    onRemoveQuestion,
    handleBack,
  } = useComposeQuestion(quiz, questionIndex);

  const { control } = methods || {};

  const { fields, remove, append } = useFieldArray({
    control,
    name: 'answers',
  });

  const { t } = useBaseHook();
  const theme = useTheme();
  const { colors } = theme;
  const styles = createStyle(theme);

  const inputsYPosition = useRef({});

  const onLayoutAnswerView = (indexInputAnswer: number, y: number) => {
    inputsYPosition.current[indexInputAnswer] = y;
  };

  const onPressAddAnswer = () => {
    append({
      content: '',
      isCorrect: false,
    });
  };

  const onValidateError: ValidateComposeQuestionError = (
    type: 'question' | 'answers',
    indexInputAnswer?: number,
  ) => {
    switch (type) {
      case 'question':
        // scroll to question field
        scrollViewRef?.current?.scrollToPosition?.(0, 0, true);
        break;
      case 'answers':
        if (isNumber(indexInputAnswer)) {
          // scroll to answer field
          scrollViewRef?.current?.scrollToPosition?.(
            0,
            inputsYPosition.current[indexInputAnswer],
            true,
          );
        } else {
          // scroll to end
          scrollViewRef?.current?.scrollToEnd?.(true);
        }
        break;
    }
  };

  useBackPressListener(handleBack);

  return (
    <View style={styles.container}>
      <Header
        useI18n
        title={
          isCreatingQuestion
            ? t('quiz:add_question')
            : t('quiz:edit_the_question_number', {
              number: questionIndex + 1,
            })
        }
        buttonProps={{
          disabled: !enabledBtnSave,
          loading,
          style: styles.btnSave,
        }}
        buttonText="common:btn_save"
        onPressButton={onSave(onValidateError)}
        onPressBack={handleBack}
      />
      <KeyboardAwareScrollView
        ref={scrollViewRef}
        style={styles.scrollView}
        contentContainerStyle={styles.contentContainer}
        alwaysBounceVertical={false}
        keyboardShouldPersistTaps="handled"
      >
        <FormProvider {...methods as any}>
          <QuestionField
            remove={onRemoveQuestion}
            questionIndex={
              isCreatingQuestion ? questions.length : questionIndex
            }
          />
          {fields.map((field, index) => (
            <AnswerField
              key={`answer_field.${field.id}`}
              remove={remove}
              answerIndex={index}
              onLayoutAnswerView={onLayoutAnswerView}
            />
          ))}
          {fields.length < MAX_ANSWERS && (
            <Button.Neutral
              type="ghost"
              icon="Plus"
              iconSize={16}
              useI18n
              style={styles.btnAddAnswer}
              onPress={onPressAddAnswer}
            >
              quiz:add_answer
            </Button.Neutral>
          )}
          {needToChooseCorrectAnswer && (
            <View style={styles.errorView}>
              <Icon
                icon="CircleExclamation"
                size={14}
                tintColor={colors.red40}
              />
              <Text.BodyXS style={styles.errorText} useI18n>
                quiz:please_select_correct_answer
              </Text.BodyXS>
            </View>
          )}
        </FormProvider>
      </KeyboardAwareScrollView>
    </View>
  );
};

const createStyle = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: colors.white,
    },
    btnSave: {
      marginRight: spacing.margin.small,
    },
    scrollView: {
      flex: 1,
    },
    contentContainer: {
      padding: spacing.padding.large,
      paddingBottom: spacing.padding.large + 84,
    },
    btnAddAnswer: {
      alignSelf: 'baseline',
      marginTop: spacing.margin.small,
    },
    errorView: {
      flexDirection: 'row',
      flex: 1,
      marginTop: spacing.margin.small,
    },
    errorText: {
      color: colors.red40,
      marginLeft: spacing.margin.tiny,
    },
  });
};

export default ComposeQuestion;
