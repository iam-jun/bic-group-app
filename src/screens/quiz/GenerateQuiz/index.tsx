export { default as SubmitFormQuiz } from './SubmitFormQuiz';
export { default as ComposeQuiz } from './ComposeQuiz';
