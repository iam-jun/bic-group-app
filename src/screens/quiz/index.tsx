export { EntryQuiz, PublishQuiz } from './SubmitFormQuiz';
export { default as ComposeQuiz } from './ComposeQuiz';
export { default as EditQuestion } from './EditQuestion';
export { default as TakeQuiz } from './TakeQuiz';
export { default as TakeQuizReview } from './TakeQuizReview';
export { default as TakeQuizResult } from './TakeQuizResult';
export { default as PreviewDraftQuizNotification } from './PreviewDraftQuizNotification';
export { default as YourQuiz } from './YourQuiz';
