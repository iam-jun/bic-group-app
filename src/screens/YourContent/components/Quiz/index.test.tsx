import React from 'react';
import { postWithQuiz } from '~/test/mock_data/quiz';
import useDraftQuizStore, { IDraftQuizState } from './store';
import { IPost } from '~/interfaces/IPost';
import { renderWithRedux } from '~/test/testUtils';
import Quiz from '.';

describe('Quiz component', () => {
  it('render correctly', () => {
    useDraftQuizStore.setState((state: IDraftQuizState) => {
      state.draftQuiz.data = [postWithQuiz] as IPost[];
      return state;
    });

    const onScroll = jest.fn();

    const rendered = renderWithRedux(<Quiz
      onScroll={onScroll}
    />);
    const contentView = rendered.getByTestId('quiz.list');

    expect(rendered.toJSON()).toMatchSnapshot();
    expect(contentView).toBeDefined();
  });

  it('should render empty view when draft quizzes are empty end cant load more', () => {
    useDraftQuizStore.setState((state: IDraftQuizState) => {
      state.draftQuiz.data = [] as IPost[];
      state.draftQuiz.hasNextPage = false;
      return state;
    });

    const onScroll = jest.fn();

    const rendered = renderWithRedux(<Quiz
      onScroll={onScroll}
    />);

    const emptyView = rendered.getByTestId('draft_quiz.empty_view');
    expect(emptyView).toBeDefined();
  });
});
