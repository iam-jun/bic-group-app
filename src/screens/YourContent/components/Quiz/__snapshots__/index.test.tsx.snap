// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quiz component render correctly 1`] = `
<RCTScrollView
  CellRendererComponent={[Function]}
  ItemSeparatorComponent={[Function]}
  ListEmptyComponent={[Function]}
  ListFooterComponent={[Function]}
  ListHeaderComponent={[Function]}
  animatedStyle={
    Object {
      "value": Object {},
    }
  }
  collapsable={false}
  data={
    Array [
      Object {
        "code": "api.ok",
        "data": Object {
          "actor": Object {
            "avatar": "https://media.beincom.io/image/variants/user/avatar/7a9ea089-12a6-40e9-96ca-083bcbc5408a",
            "email": "truongthi@evol.vn",
            "fullname": "Nguyễn Trường Thi",
            "id": "0fa01fde-7c15-4d55-b60a-8e990123bc2e",
            "isDeactivated": false,
            "isVerified": true,
            "showingBadges": Array [
              Object {
                "community": Object {
                  "id": "656cebfe-1b91-473f-97fd-96837bf9e2a5",
                  "name": "Community của Bảo gất là dễ thương nuônnnnnnnnnnnnnn",
                },
                "iconUrl": "https://media.beincom.io/image/variants/badge/a2a4b6cd-4581-42e6-b28c-dd9d04fec9a6",
                "id": "61a0b534-c390-4685-936e-621e32607953",
                "name": "Team Secret",
              },
            ],
            "username": "truongthi",
          },
          "audience": Object {
            "groups": Array [
              Object {
                "communityId": "15337361-1577-4b7b-a31d-990df06aa446",
                "icon": "https://media.beincom.io/image/variants/group/avatar/6f4e6113-fd61-4b5b-848b-468afe2c8550",
                "id": "18508ac3-2bfc-4172-b071-1d67f1b1e05b",
                "isCommunity": false,
                "name": "Bein Product Team",
                "privacy": "OPEN",
                "rootGroupId": "35b5fb8f-6f7a-4ac2-90bb-18199096c429",
              },
              Object {
                "communityId": "15337361-1577-4b7b-a31d-990df06aa446",
                "icon": null,
                "id": "aeab68c2-bcec-4edb-a78b-60c0ee90afd7",
                "isCommunity": false,
                "name": "Bein Founding Community",
                "privacy": "OPEN",
                "rootGroupId": "35b5fb8f-6f7a-4ac2-90bb-18199096c429",
              },
              Object {
                "communityId": "153c02e4-3b08-4c3e-bb74-dd2d94d0a5e7",
                "icon": "https://media.beincom.io/image/variants/group/avatar/88939640-cfdb-43ac-8689-9bc6e584bc9a",
                "id": "b01fb58e-9299-4a0e-a55f-9839293fb42a",
                "isCommunity": true,
                "name": "Community test group order Linh",
                "privacy": "PRIVATE",
                "rootGroupId": "b01fb58e-9299-4a0e-a55f-9839293fb42a",
              },
            ],
          },
          "commentsCount": 0,
          "communities": Array [
            Object {
              "communityId": "15337361-1577-4b7b-a31d-990df06aa446",
              "icon": "https://media.beincom.io/image/variants/group/avatar/26a806d3-3557-4f7a-b96a-132a9befccff",
              "id": "35b5fb8f-6f7a-4ac2-90bb-18199096c429",
              "isCommunity": true,
              "name": "EVOL Community",
              "privacy": "OPEN",
              "rootGroupId": "35b5fb8f-6f7a-4ac2-90bb-18199096c429",
            },
            Object {
              "communityId": "153c02e4-3b08-4c3e-bb74-dd2d94d0a5e7",
              "icon": "https://media.beincom.io/image/variants/group/avatar/88939640-cfdb-43ac-8689-9bc6e584bc9a",
              "id": "b01fb58e-9299-4a0e-a55f-9839293fb42a",
              "isCommunity": true,
              "name": "Community test group order Linh",
              "privacy": "PRIVATE",
              "rootGroupId": "b01fb58e-9299-4a0e-a55f-9839293fb42a",
            },
          ],
          "content": "# Người dùng vẫn sạc smartphone qua đêm
Khảo sát của Asus cho thấy người dùng thường xuyên sạc điện thoại qua đêm dù thói quen này được đánh giá gây hại cho thiết bị.
Asus cho biết một nửa người dùng smartphone của hãng chọn sạc qua đêm, dù thiết bị đã có tính năng sạc nhanh. Trong đó, 25% bật Sạc ổn định - tính năng làm chậm tốc độ sạc để giảm nhiệt và kéo dài tuổi thọ pin; 26% bật Sạc theo lịch trình - chế độ trì hoãn việc sạc đầy 100% pin.Trang công nghệ _Android Authority_ cũng thực hiện khảo sát tương tự và có tới 66% người tham gia có thói quen cắm sạc thiết bị cả đêm.",
          "createdAt": "2023-06-29T05:24:25.545Z",
          "id": "54f4a2eb-034d-4e4e-8810-44744bffc87d",
          "isReported": false,
          "isSaved": false,
          "linkPreview": null,
          "markedReadPost": false,
          "media": Object {
            "files": Array [],
            "images": Array [],
            "videos": Array [],
          },
          "mentions": Array [],
          "ownerReactions": Array [
            Object {
              "id": "427b3794-aa51-4bb6-8ddb-743861585f90",
              "reactionName": "mask",
            },
          ],
          "privacy": "OPEN",
          "quiz": Object {
            "createdAt": "2023-06-29T08:37:43.927Z",
            "description": "ahihihi",
            "id": "c14e8f68-2b33-4303-ab81-0f29bb3192ac",
            "status": "PUBLISHED",
            "title": "testse",
            "updatedAt": "2023-06-29T08:38:39.825Z",
          },
          "reactionsCount": Array [
            Object {
              "mask": 1,
            },
          ],
          "setting": Object {
            "canComment": true,
            "canReact": true,
            "importantExpiredAt": null,
            "isImportant": false,
          },
          "status": "PUBLISHED",
          "tags": Array [],
          "totalUsersSeen": 3,
          "type": "POST",
        },
        "meta": Object {
          "message": "OK",
        },
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onEndReached={[Function]}
  onEndReachedThreshold={0.2}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  refreshControl={
    <RefreshControlMock
      onRefresh={[Function]}
      progressViewOffset={48}
      refreshing={true}
      tintColor="#A4A9B7"
    />
  }
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  testID="quiz.list"
  viewabilityConfigCallbackPairs={Array []}
>
  <RCTRefreshControl />
  <View>
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Object {
            "paddingTop": 48,
          }
        }
      >
        <View
          style={
            Object {
              "height": 16,
              "paddingHorizontal": undefined,
              "paddingVertical": undefined,
              "width": undefined,
            }
          }
        />
      </View>
    </View>
    <View
      animatedStyle={
        Object {
          "value": Object {},
        }
      }
      collapsable={false}
      onLayout={[Function]}
    >
      <View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#FFFFFF",
              },
              undefined,
            ]
          }
        >
          <View
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flexDirection": "row",
                "marginHorizontal": 16,
                "opacity": 1,
                "paddingTop": 8,
              }
            }
            testID="content_header"
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "marginRight": 12,
                  "marginTop": 4,
                  "opacity": 1,
                }
              }
              testID="button_wrapper"
            >
              <View
                style={
                  Array [
                    Object {
                      "borderRadius": 8,
                      "height": 48,
                      "width": 48,
                    },
                    undefined,
                  ]
                }
                testID="avatar"
              >
                <View
                  style={
                    Array [
                      Array [
                        Object {
                          "borderRadius": 8,
                          "height": 48,
                          "width": 48,
                        },
                        Object {
                          "borderRadius": 24,
                        },
                      ],
                      Object {
                        "backgroundColor": "#A4A9B7",
                      },
                    ]
                  }
                  testID="avatar_container"
                >
                  <View
                    style={
                      Array [
                        Object {
                          "overflow": "hidden",
                        },
                        Array [
                          Array [
                            Array [
                              Object {
                                "borderRadius": 8,
                                "height": 48,
                                "width": 48,
                              },
                              Object {
                                "borderRadius": 24,
                              },
                            ],
                            Object {
                              "borderColor": "#F2F5F8",
                              "borderWidth": 2,
                            },
                            Object {
                              "backgroundColor": "#FFFFFF",
                            },
                          ],
                          false,
                        ],
                      ]
                    }
                  >
                    <FastImageView
                      onFastImageError={[Function]}
                      resizeMode="cover"
                      source={
                        Object {
                          "testUri": "../../../assets/images/user_avatar_default.png",
                        }
                      }
                      style={
                        Object {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                      testID="avatar.image"
                    />
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "marginRight": 12,
                    "opacity": 1,
                  }
                }
                testID="button_wrapper"
              >
                <Text
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "color": "#212745",
                        "fontFamily": "BeVietnamPro-Medium",
                        "fontSize": 15,
                      },
                      Object {
                        "color": "#181C32",
                      },
                      Object {
                        "flexShrink": 1,
                      },
                    ]
                  }
                  testID="content_header.actor"
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "marginVertical": 4,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#212745",
                        "fontFamily": "BeVietnamPro-Light",
                        "fontSize": 13,
                      },
                      Object {
                        "color": "#353E6E",
                      },
                      Object {
                        "marginRight": 4,
                      },
                    ]
                  }
                >
                  to
                </Text>
                <Text
                  onPress={[Function]}
                  style={
                    Array [
                      Object {
                        "color": "#212745",
                        "fontFamily": "BeVietnamPro-Medium",
                        "fontSize": 13,
                      },
                      Object {
                        "color": "#181C32",
                      },
                      undefined,
                    ]
                  }
                  testID="content_header.audiences"
                >
                  
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#212745",
                        "fontFamily": "BeVietnamPro-Light",
                        "fontSize": 11,
                      },
                      Object {
                        "color": "#353E6E",
                      },
                      undefined,
                    ]
                  }
                  testID="time_view"
                >
                  
                </Text>
              </View>
            </View>
          </View>
          <View>
            <View
              style={
                Object {
                  "marginHorizontal": 16,
                  "marginVertical": 8,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                testID="post_view_content"
              >
                <View
                  style={false}
                >
                  <View />
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "paddingBottom": 16,
                "paddingHorizontal": 16,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignSelf": "stretch",
                    "backgroundColor": "#E3E6F2",
                    "height": 1,
                    "marginHorizontal": 0,
                    "width": undefined,
                  },
                  Object {},
                ]
              }
              testID="divider"
            />
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "paddingBottom": 16,
                "paddingLeft": 16,
              }
            }
          >
            <View
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "borderRadius": 8,
                  },
                  undefined,
                ]
              }
              testID="quiz_draft_view.button_delete"
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 8,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "paddingVertical": 4,
                    },
                    undefined,
                    Object {
                      "backgroundColor": "#FEECED",
                      "height": 36,
                      "paddingHorizontal": 12,
                    },
                  ]
                }
                testID="button.content"
              >
                <View
                  accessibilityState={
                    Object {
                      "disabled": true,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={false}
                  hitSlop={
                    Object {
                      "bottom": 10,
                      "left": 10,
                      "right": 10,
                      "top": 10,
                    }
                  }
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                        },
                        undefined,
                        Object {
                          "backgroundColor": undefined,
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          undefined,
                          undefined,
                          undefined,
                        ]
                      }
                    >
                      <View
                        testID="button.icon"
                      />
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "height": undefined,
                  "paddingHorizontal": undefined,
                  "paddingVertical": undefined,
                  "width": 16,
                }
              }
            />
            <View
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "borderRadius": 8,
                  },
                  undefined,
                ]
              }
              testID="quiz_draft_view.button_edit"
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 8,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "paddingVertical": 4,
                    },
                    undefined,
                    Object {
                      "backgroundColor": "#EBF4FF",
                      "height": 36,
                      "paddingHorizontal": 12,
                    },
                  ]
                }
                testID="button.content"
              >
                <View
                  accessibilityState={
                    Object {
                      "disabled": true,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={false}
                  hitSlop={
                    Object {
                      "bottom": 10,
                      "left": 10,
                      "right": 10,
                      "top": 10,
                    }
                  }
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                        },
                        undefined,
                        Object {
                          "backgroundColor": undefined,
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          undefined,
                          undefined,
                          undefined,
                        ]
                      }
                    >
                      <View
                        testID="button.icon"
                      />
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "height": 8,
                "paddingHorizontal": undefined,
                "paddingVertical": undefined,
                "width": undefined,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Object {
            "height": 16,
            "paddingHorizontal": undefined,
            "paddingVertical": undefined,
            "width": undefined,
          }
        }
      />
    </View>
  </View>
</RCTScrollView>
`;
